pkg/version/release.go

```diff
@@ -25,22 +27,25 @@ func ParseGitVersion(gitVersion string) (*ReleaseVersion, error) {
 	}, nil
 }
 
-// FirstMinorRelease returns the minor release but the patch releases always be 0(vx.y.0). e.g:
-// - v1.2.1-12-g2eb92858 --> v1.2.0
-// - v1.2.3-12-g2e860210 --> v1.2.0
-func (r *ReleaseVersion) FirstMinorRelease() string {
+// ReleaseVersion returns the current version with format "vx.y.z".
+// It could be patch release or pre-release
+func (r *ReleaseVersion) ReleaseVersion() string {
 	if r.Version == nil {
 		return "<nil>"
 	}
 
-	return fmt.Sprintf("v%d.%d.0", r.Version.Major(), r.Version.Minor())
+	return fmt.Sprintf("v%s", r)
 }
 
-// PatchRelease returns the stable version with format "vx.y.z".
-func (r *ReleaseVersion) PatchRelease() string {
-	if r.Version == nil {
-		return "<nil>"
-	}
-
-	return fmt.Sprintf("v%d.%d.%d", r.Version.Major(), r.Version.Minor(), r.Version.Patch())
+// removeGitVersionCommits removes the git commit info from the version
+// The git version looks like: v1.0.4-14-g2414721
+// the current head of my "parent" branch is based on v1.0.4,
+// but since it has a few commits on top of that, describe has added the number of additional commits ("14")
+// and an abbreviated object name for the commit itself ("2414721") at the end.
+func removeGitVersionCommits(gitVersion string) string {
+	// This match the commit info part of the git version
+	splitRE := regexp.MustCompile("-[0-9]+-g[0-9a-z]{7}")
+	match := splitRE.Split(gitVersion, 2)
+
+	return match[0]
```

Comment:
Can we ensure the `len(match)` is always not `nil`? 
I‘m concerned that there is a panic here for abnormal inputs.

Reply:
I tried with the following code, there is no panic, and work as expected:
```go
package main

import (
	"fmt"
	"regexp"
)

func removeGitVersionCommits(v string) string {
	splitRE := regexp.MustCompile("-[0-9]+-g[0-9a-z]{7}")
	matches := splitRE.Split(v, 2)

	fmt.Println(matches)
	return matches[0]
}

func main() {
	fmt.Println(removeGitVersionCommits(""))
}
```
The output:
```sh
[]
```

Reply:
OK.